sympy/matrices/expressions/hadamard.py

```diff
@@ -22,7 +22,7 @@ def hadamard_product(*matrices):
     >>> hadamard_product(A)
     A
     >>> hadamard_product(A, B)
-    A.*B
+    HadamardProduct(A, B)
```

Comment:
Omg... `A.*B` is a lot nicer. I'm giving -1 to this PR.

Reply:
I would recommend restoring `A.*B`

Reply:
Maybe you should edit the Python printer instead of the StrPrinter? Python code printer should return `HadamardProduct`, I agree.

Reply:
That's why we have the pretty printer. We've had this restriction on the str printer for some time, even if many times it isn't always followed.

Reply:
The pretty printer is not the default output. Also consider that the NumPy bug would have not been detected, had this been printed in a rebuildable way.

Reply:
You mean the original lambdify issue? I think it would have, because HadamardProduct would be undefined. 

> The pretty printer is not the default output.

It is for the notebook and isympy, which are what we recommend people use for interactive SymPy usage.

Reply:
Although the original reason for this PR is the comment in #17138, I would expect that editing in https://github.com/spyder-ide/spyder-kernels/pull/129 doesn't work for all functions that have an unreproducable str-printing. Also, if one would like to manually edit a large expression that you almost got correct, this is crucial. `srepr` is a bit too complicated to be manageable.